lib/diaspora_federation/entity.rb

```diff
@@ -319,8 +319,8 @@ def simple_node(doc, name, value)
       when :integer
         text.to_i if text =~ /^\d+$/
       when :boolean
-        return true if text =~ /(true|t|yes|y|1)$/i
-        false if text =~ /(false|f|no|n|0)$/i
+        return true if text =~ /^(true|t|yes|y|1)$/i
+        false if text =~ /^(false|f|no|n|0)$/i
```

Comment:
Is it a real issue here? We can declare multiline value for booleans, integers and timestamps as undefined behavior. I'm not sure one would send multiline strings here and expect some determined results. Perhaps, we could add a validation for a single-line value instead for these types.